### ISSUE TITLE
Unable to ingest objects

### ISSUE THREAD

**Issue by bseeger** [mjordan/islandora_workbench#96]
Hi Mark, 

I'm now checking out the workbench.  When using it, everything looks good until I get to the point of running: 

` ./workbench --config create.yml`

I get no output on the command line and some error messages in the log: 

```
13-Apr-20 15:40:29 - INFO - "Create" task started using config file create.yml
13-Apr-20 15:40:31 - WARNING - Node for CSV record 01 not created, HTTP response code was 406.
13-Apr-20 15:40:31 - WARNING - Node for CSV record 02 not created, HTTP response code was 406.
13-Apr-20 15:40:31 - WARNING - Node for CSV record 03 not created, HTTP response code was 406.
13-Apr-20 15:40:31 - WARNING - Node for CSV record 04 not created, HTTP response code was 406.
13-Apr-20 15:40:31 - WARNING - Node for CSV record 05 not created, HTTP response code was 406.
```

I added a few debug statements: 

```
13-Apr-20 15:59:32 - WARNING - Node for CSV record 03 not created, HTTP response code was 406.
13-Apr-20 15:59:32 - WARNING - POSTING
13-Apr-20 15:59:32 - WARNING - RESP: {"message":"No route found for the specified format json. Supported formats: html."}
13-Apr-20 15:59:32 - WARNING - Headers: {'Date': 'Mon, 13 Apr 2020 19:59:32 GMT', 'Server': 'Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/7.2.29', 'X-Content-Type-Options': 'nosniff, nosniff', 'X-Powered-By': 'PHP/7.2.29', 'Link': '<http://example.com:8000/node?_format=html>; rel="alternate"; type="text/html"', 'Cache-Control': 'must-revalidate, no-cache, private', 'X-UA-Compatible': 'IE=edge', 'Content-language': 'en', 'X-Frame-Options': 'SAMEORIGIN', 'Expires': 'Sun, 19 Nov 1978 05:00:00 GMT', 'Vary': '', 'X-Generator': 'Drupal 8 (https://www.drupal.org)', 'Content-Length': '84', 'Keep-Alive': 'timeout=5, max=99', 'Connection': 'Keep-Alive', 'Content-Type': 'application/json'}
```

Seems like the JSON API is not working correctly? I did enable the JSON: API module under Extend, but maybe I missed something?

**Comment by bseeger** [comment]
I should mention, I'm just running with your sample data alone - I haven't changed anything except the URL and credentials in the create.yml file.

**Comment by mjordan** [comment]
@bseeger can you run with `--check` (i.e., `./workbench --config create.yml --check`) and let me know what the console and the log say?

**Comment by mjordan** [comment]
That "Link': '<http://example.com:8000/node?_format=html>" looks very strange...

**Comment by mjordan** [comment]
Based on what your debug logging revealed, I'm guessing you need to follow some configuration instructions that I was sure were no longer necessary... sorry about that. Here are those instructions:

* You must enable the following two REST Resources at `admin/config/services/rest`:
   * Field
    * Field storage
      * For these two resources, set "Granularity" to "Method" and check "GET", "Accepted request formats" to "JSON", and "Authentication providers" to "basic_auth".

**Comment by mjordan** [comment]
Hm, I just spun up a fresh islandora playbook and used workbench to ingest the sample data. The only config I did was enable the JSON:API module. I didn't have to enable the two REST resources I mentioned in my previous comment.

**Comment by bseeger** [comment]
Hi Mark, this is what I've gotten with check: 
```
$ ./workbench --config create.yml --check
OK, configuration file has all required values (did not check for optional values).
OK, http://msel-example.mse.jhu.edu:8000/ is accessible using the credentials provided.
OK, CSV file input_data/metadata.csv found.
OK, all 5 rows in the CSV file have the same number of columns as there are headers (5).
OK, CSV column headers match Drupal field names.
OK, required Drupal fields are present in the CSV file.
OK, term IDs in CSV file exist in their respective taxonomies.
OK, files named in the CSV "file" column are all present.
Configuration and input data appear to be valid.
```
So things look good.  FYI -- I am working with a remote server on a VPN.

**Comment by bseeger** [comment]
Error seen in drupal: 
```
Symfony\Component\HttpKernel\Exception\NotAcceptableHttpException: No route found for the specified format json. Supported formats: html. in Drupal\Core\Routing\RequestFormatRouteFilter->filter() (line 65 of /var/www/html/drupal/web/core/lib/Drupal/Core/Routing/RequestFormatRouteFilter.php).```

**Comment by bseeger** [comment]
Is this the right spot to enable the JSON API?  
![Screen Shot 2020-04-14 at 9 06 51 AM](https://user-images.githubusercontent.com/7966375/79228400-5736e700-7e2f-11ea-9c87-60197415e931.png)

**Comment by bseeger** [comment]
![Screen Shot 2020-04-14 at 9 08 21 AM](https://user-images.githubusercontent.com/7966375/79228489-82b9d180-7e2f-11ea-8735-f9415e1b562a.png)

**Comment by bseeger** [comment]
![Screen Shot 2020-04-14 at 9 16 14 AM](https://user-images.githubusercontent.com/7966375/79229185-9b76b700-7e30-11ea-9921-b662c85ec1a6.png)

**Comment by mjordan** [comment]
Strange that Drupal says you have JSON enabled yet it's complaining that only HTML is. The fact that your output from `--check` is good means that JSON is in fact enabled for the JSON:API (that is what Workbench queries to get Drupal field definitions). It sounds like it's the `node` endpoint that is throwing that error.

To confirm, can you try `POST`ing to that endpoint and see what happens? Try this:

`curl -v -i -X POST -u admin:islandora -H"Content-type: application/json" --data '{"type":[{"target_id":"islandora_object","target_type":"node_type"}],"title":[{"value":"Created a test collection with POST"}],"field_model":[{"target_id":23,"target_type":"taxonomy_term"}]}' 'http://localhost:8000/node?_format=json'`

With the URL and credentials of your site. We'll want the response code and response body text, plus check to see if that "NotAcceptableHttpException"  error is being written to your Drupal/apache error log.

**Comment by bseeger** [comment]
That worked just fine and the new collection was created.  No error in the log file. 

```
$ curl -v -i -X POST -u admin:islandora -H"Content-type: application/json" --data '{"type":[{"target_id":"islandora_object","target_type":"node_type"}],"title":[{"value":"Created a test collection with POST"}],"field_model":[{"target_id":23,"target_type":"taxonomy_term"}]}' 'http://msel-example.mse.jhu.edu:8000/node?_format=json' -u admin:password

* About to connect() to msel-islandora01.mse.jhu.edu port 8000 (#0)
*   Trying 10.162.127.39...
* Connected to msel-islandora01.mse.jhu.edu (10.162.127.39) port 8000 (#0)
* Server auth using Basic with user 'admin'
> POST /node?_format=json HTTP/1.1
> Authorization: Basic YWRtaW46Q3Nna2VSSlZ3NmN5amhGcGtl
> User-Agent: curl/7.29.0
> Host: msel-islandora01.mse.jhu.edu:8000
> Accept: */*
> Content-type: application/json
> Content-Length: 190
>
* upload completely sent off: 190 out of 190 bytes
< HTTP/1.1 201 Created
HTTP/1.1 201 Created
< Date: Tue, 14 Apr 2020 14:18:08 GMT
Date: Tue, 14 Apr 2020 14:18:08 GMT
< Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/7.2.29
Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/7.2.29
< X-Content-Type-Options: nosniff
X-Content-Type-Options: nosniff
< X-Powered-By: PHP/7.2.29
X-Powered-By: PHP/7.2.29
< Location: http://msel-islandora01.mse.jhu.edu:8000/node/151
Location: http://msel-islandora01.mse.jhu.edu:8000/node/151
< Cache-Control: must-revalidate, no-cache, private
Cache-Control: must-revalidate, no-cache, private
< X-UA-Compatible: IE=edge
X-UA-Compatible: IE=edge
< Content-language: en
Content-language: en
< X-Content-Type-Options: nosniff
X-Content-Type-Options: nosniff
< X-Frame-Options: SAMEORIGIN
X-Frame-Options: SAMEORIGIN
< Expires: Sun, 19 Nov 1978 05:00:00 GMT
Expires: Sun, 19 Nov 1978 05:00:00 GMT
< Vary:
Vary:
< X-Generator: Drupal 8 (https://www.drupal.org)
X-Generator: Drupal 8 (https://www.drupal.org)
< Content-Length: 2033
Content-Length: 2033
< Content-Type: application/json
Content-Type: application/json

<
{"nid":[{"value":151}],"uuid":[{"value":"ac7b9e28-6c4a-4dab-8711-2b1e4f6505ac"}],"vid":[{"value":154}],"langcode":[{"value":"en"}],"type":[{"target_id":"islandora_object","target_type":"node_type","target_uuid":"9a62e804-e25a-4f72-8b57-4c82bb1f0812"}],"revision_timestamp":[{"value":"2020-04-14T14:18:08+00:00","format":"Y-m-d\\TH:i:sP"}],"revision_uid":[{"target_id":1,"target_type":"user","target_uuid":"fe0db321-df08-4317-bb42-f6180122fdb4","url":"\/user\/1"}],"revision_log":[],"status":[{"value":true}],"uid":[{"target_id":1,"target_type":"user","target_uuid":"fe0db321-df08-4317-bb42-f6180122fdb4","url":"\/user\/1"}],"title":[{"value":"Created a test collection with POST"}],"created":[{"value":"2020-04-14T14:18:08+00:00","format":"Y-m-d\\TH:i:sP"}],"changed":[{"value":"2020-04-14T14:18:08+00:00","format":"Y-m-d\\TH:i:sP"}],"promote":[{"value":true}],"sticky":[{"value":false}],"default_langcode":[{"value":true}],"revision_translation_affected":[{"value":true}],"content_translation_source":[{"value":"und"}],"con* Connection #0 to host msel-islandora01.mse.jhu.edu left intact
tent_translation_outdated":[{"value":false}],"field_access_terms":[],"field_alternative_title":[],"field_classification":[],"field_coordinates":[],"field_coordinates_text":[],"field_description":[],"field_dewey_classification":[],"field_display_hints":[],"field_edition":[],"field_edtf_date":[],"field_edtf_date_created":[],"field_edtf_date_issued":[],"field_extent":[{"value":"1 item"}],"field_genre":[],"field_geographic_subject":[],"field_identifier":[],"field_isbn":[],"field_language":[],"field_lcc_classification":[],"field_linked_agent":[],"field_local_identifier":[],"field_member_of":[],"field_model":[{"target_id":23,"target_type":"taxonomy_term","target_uuid":"677f1548-bd85-403e-aed7-89c7bfdbd996","url":"\/taxonomy\/term\/23"}],"field_note":[],"field_oclc_number":[],"field_physical_form":[],"field_pid":[],"field_place_published":[],"field_resource_type":[],"field_rights":[],"field_subject":[],"field_subjects_name":[],"field_table_of_contents":[],"field_temporal_subject":[],"field_weight":[]}

**Comment by mjordan** [comment]
OK, thanks, at least that narrows it down a bit (I think). I'm heading into some meetings next few hours but if you have time, can you log the arguments to the `issue_request` function (https://github.com/mjordan/islandora_workbench/blob/master/workbench_utils.py#L105) to see if we can pin down what request is producing the `406` error?

**Comment by bseeger** [comment]
Oh! I figured it out.... if you leave the forward slash on the end of the url in the config file, you will see this error.  

```
task: create
host: "http://msel-example.mse.jhu.edu:8000/"
```
I removed that slash and things ingested just fine.

**Comment by mjordan** [comment]
Nice! Sorry to put you through that. I'll remove it in the code and credit you. Thanks! Let me know if you hit anything else.

**Comment by bseeger** [comment]
Thanks for the help, @mjordan!

**Comment by mjordan** [comment]
Fixed with f9cc8ebd77428b462166cac3d3975500a45e8a34. Thanks!

**Comment by bseeger** [comment]
Thanks, @mjordan
